### ISSUE TITLE
Code Table Request - Add ectoparasite: Diptera to Examined_Detected Code Table

### ISSUE THREAD

**Issue by campmlc** [ArctosDB/arctos#6502]
## Goal: 

Add  ectoparasite: Diptera to the Examined_Detected code table:

## Proposed Values: 

ectoparasite: ~~Diptera~~fly

## Proposed Definitions: 

~~"Ectoparasitic Diptera, includes bat flies, bot flies and other hippoboscidae. Avoid using for dipteran larvae that are found on dead organisms, unless these were known to be feeding on the organism while it was alive (myiasis)."~~
Flies are insects in the order Diptera. Specify specific type (e.g. bat flies, bot flies, etc) and life stage if known in methods/remarks.

## Context: 

These values are being added the Examined_Detected code table to capture existing data in Arctos, to accommodate ongoing research which will be populating these attributes going forward, and for TPT digitization and host linkage. The current attributes "ectoparasite examination" and "ectoparasite detected" will be migrated to these new attributes. Splitting from #6260 .

## Table: 

~~https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_code_tables~~
https://arctos.database.museum/info/ctDocumentation.cfm?table=ctexamined_detected

## Priority: 

High. Needed for TPT Arctos and MSB funding, and for upcoming presentation at American Society of Parasitology meetings July 13-15, 2023.

## Available for Public View: 

Yes

## Project: 

Add the issue to the [Code Table Management Project](https://github.com/ArctosDB/arctos/projects/13#card-31628184)._

## Discussion:
 
@jldunnum @campmlc @acdoll @bryansmclean 

@ArctosDB/arctos-code-table-administrators 

## Approval

_All of the following must be checked before this may proceed._

_The [How-To Document](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html) should be followed. Pay particular attention to terminology (with emphasis on consistency) and documentation (with emphasis on functionality)._

- [x] Code Table Administrator[1] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [x] Code Table Administrator[2] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [x] DBA - The request is functionally acceptable. The term is not a functional duplicate, and is compatible with existing data and code.
- [x] DBA - Appropriate code or handlers are in place as necessary. (ID_References, Media Relationships, Encumbrances, etc. require particular attention)

## Rejection

_If you believe this request should not proceed, explain why here. Suggest any changes that would make the change acceptable, alternate (usually existing) paths to the same goals, etc._

1. _Can a suitable solution be found here? If not, proceed to (2)_
2. _Can a suitable solution be found by Code Table Committee discussion? If not, proceed to (3)_
3. _Take the discussion to a monthly Arctos Working Group meeting for final resolution._

## Implementation

_Once all of the Approval Checklist is appropriately checked and there are no Rejection comments, or in special circumstances by decree of the Arctos Working Group, the change may be made._

_Review everything one last time.  Ensure the How-To has been followed. Ensure all checks have been made by appropriate personnel._

_Make changes as described above. Ensure the URL of this Issue is included in the definition._

_Close this Issue._

_**DO NOT** modify Arctos Authorities in any way before all points in this Issue have been fully addressed; data loss may result._

## Special Exemptions

_In very specific cases and by prior approval of The Committee, the approval process may be skipped, and implementation requirements may be slightly altered. Please note here if you are proceeding under one of these use cases._

1. _Adding an existing term to additional collection types may proceed immediately and without discussion, but doing so may also subject users to future cleanup efforts. If time allows, please review the term and definition as part of this step._
2. _The Committee may grant special access on particular tables to particular users. This should be exercised with great caution only after several smooth test cases, and generally limited to "taxonomy-like" data such as International Commission on Stratigraphy terminology._

**Comment by Jegelewicz** [comment]
Changed the code table to https://arctos.database.museum/info/ctDocumentation.cfm?table=ctexamined_detected

**Comment by Jegelewicz** [comment]
I am on the fence about this one - shouldn't we use fly to be consistent with flea, tick, mite?

**Comment by dustymc** [comment]
>use fly 

Makes sense.

I dislike the definition, it's too method-ey - I'm not sure there's any test with the kind of specificity rejecting postmortem arrivals would require (and if so it surely requires a methodology explanation!).

I separately dislike "hippoboscidae" in the definition (and a bunch of these have a similar problem) - that implies some sort of circumscription, those are complex and/or dynamic, if that level of detail is necessary then this is probably just the wrong tool for the job. It somehow feels a bit heretic to suggest such things, but I think more general definitions ("fly"==>"flies and things that pretend to be...") may be more appropriate here.

As always, data would be very useful.

**Comment by campmlc** [comment]
I was considering that - but fly is a much less specific term even as a common name than the others, which are clearly associated with specific taxonomic groups, and it seemed more likely to be abused or confused with something inappropriate. Also, to me at least, the order Diptera is much more widely understood than the terms for the other groups. This is mostly going to be used with the "Detected" attribute - meaning the paired terms will be examined for: ectoparasite and Detected: ectoparasite: Diptera vs ectoparasite: fly. I could go either way. 

For definition, something simpler?

ectoparasite: fly  
definition: flies (Diptera) ectoparasitic as adults or larvae. Specify specific type (e.g. bat flies, bot flies, etc) and life stage if known in methods/remarks.

Here are some search results:
1) search on "bat fl" in park remarks" https://arctos.database.museum/search.cfm?customoidoper=LIST&part_search=bat%20fl
Note that these all need significant work to capture remarks data on parasites from various fields into some standard format based on our new model.

[ArctosDataXG8XBe8mDF.zip](https://github.com/ArctosDB/arctos/files/11972753/ArctosDataXG8XBe8mDF.zip)

2) these 3 show the challenges of what we are trying to capture. https://arctos.database.museum/search.cfm?customoidoper=LIST&id_issuedby=%3D&oidtype=NK&oidnum=%3D181981
I've got a bat host record with the "examined for" info entered in Collection Object Remarks; the "detected" info is in part remarks. 
We've cataloged the ectoparasites and linked these to the host record, but we don't have an easy way to create the reciprocal linkage from host to parasites through the current notification system.

**Comment by Jegelewicz** [comment]
> we don't have an easy way to create the reciprocal linkage from host to parasites through the current notification system.

https://arctos.database.museum/info/unreciprocated_relationships.cfm

[PendingReciprocalRelations(4).csv](https://github.com/ArctosDB/arctos/files/11972782/PendingReciprocalRelations.4.csv)

**Comment by campmlc** [comment]
Yes, but there is no way to spot check those relationships prior to load. We used to get a notification with a link to each of the proposed related guids, so we could click to see what was being linked, who created the linkage, etc. I feel very uncomfortable loading all relationships for all MSB:Mamm without any review especially without knowing whether if I or my students created them.

Just realized that the student entering these put "parasite of" as the relationship to both the host guid and also to the NK, the latter of which won't resolve.  Relationship to NK needs to be self to enable searching. I don't see a "Manage Identifiers" in Tools menu? Am I missing something?

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
> fly is a much less specific term even as a common name than the others

> Diptera is much more widely understood than the terms for the other groups.

These seem like good reasons to me to use Diptera.

**Comment by dustymc** [comment]
>something inappropriate

If there's something inappropriate here, it might be the "ectoparasite:" bit of the term - pretty sure it's a maggot, found it in my mouse, you're forcing me to either make assertions that I can't support or to toss out the data. (And maybe that's the point, I'm just trying to understand!)


>clearly associated with specific taxonomic groups

.. which change over time. There's a data object that can handle that, but it's not this one. (That said, for this purpose "Diptera" and "fly" seem wholly interchangeable to me.)

>we don't have an easy way to create the reciprocal linkage

1. You can create relationships, or
2. Arctos will give you all the data necessary to create relationships, or
3. Arctos will create the relationships for you.

This has come up a bunch of times, I just can't understand it (nor why it's here). Elaborate?

**Comment by Jegelewicz** [comment]
> Just realized that the student entering these put "parasite of" as the relationship to both the host guid and also to the NK, the latter of which won't resolve. Relationship to NK needs to be self to enable searching. I don't see a "Manage Identifiers" in Tools menu? Am I missing something?

There is no such tool and never has been. I can fix these for you if you want with a remove and bulk upload.

Here are all the NKs with a relationship other than self.
[identifierDownload(3).csv](https://github.com/ArctosDB/arctos/files/11972870/identifierDownload.3.csv)

**Comment by dustymc** [comment]
> Relationship to NK needs to be self to enable searching.

There remains some very large gap in the understanding of the identifiers model.

**Comment by Jegelewicz** [comment]
> no way to spot check those relationships prior to load. We used to get a notification with a link to each of the proposed related guids,

I added a column with a simple formula to create the links, then pasted them as links in the last column.
[PendingReciprocalRelations(4).xlsx](https://github.com/ArctosDB/arctos/files/11972899/PendingReciprocalRelations.4.xlsx)

**Comment by campmlc** [comment]
Thanks, @Jegelewicz - we need the Para record link as well. I'll file a separate issue to request this in the notification download. 
I'd be happy to sit down at some convenient time for everyone and go over the model by which NKs, AFs, IFs, etc are used in linking parasites and hosts as a common linking field identifier for MSB, UAM, HWML, and elsewhere, and have been used as such since the 1980s. But I don't want to hijack this issue. 

Re: Diptera: yes, we don't know for sure they are ectoparasitic or even flies. We also collect specs of dirt that get cataloged as ecto parts. Another reason to use this attribute as a flag prior to cataloging reciprocal relationships to real cataloged items. 

For this issue, OK to use:
ectoparasite: fly
definition: flies (Diptera) ectoparasitic as adults or larvae. Specify specific type (e.g. bat flies, bot flies, etc) and life stage if known in methods/remarks. 
?

**Comment by Jegelewicz** [comment]
> we need the Para record link as well
[PendingReciprocalRelations(4).xlsx](https://github.com/ArctosDB/arctos/files/11973010/PendingReciprocalRelations.4.xlsx)

**Comment by Jegelewicz** [comment]
Suggest

ectoparasite: fly

definition: Flies are insects in the order Diptera. Specify specific type (e.g. bat flies, bot flies, etc) and life stage if known in methods/remarks.

Also suggest we add a version of the second sentence to all the other requests...

**Comment by campmlc** [comment]
Sounds good.

**Comment by Jegelewicz** [comment]
added
